crates/frontend/src/pages/context_override.rs

```diff
@@ -172,71 +167,78 @@ pub fn context_override() -> impl IntoView {
+                Operator::Is,
+                dim.dimension.clone(),
+                r#type.unwrap(),
+            ));
+            if let Err(_) = condition {
+                //TODO emit and alert and return
+                return;
+            }
+
+            default_ctx.push(condition.unwrap());
+        }
+
+        selected_context_ws.set(Some(Data {
+            context: default_ctx,
             overrides: vec![],
         }));
         open_drawer("context_and_override_drawer");
     });
 
-    let handle_submit = Callback::new(move |_| {
+    let on_submit = Callback::new(move |_| {
         close_drawer("context_and_override_drawer");
         set_form_mode.set(None);
-        set_selected_data.set(None);
+        selected_context_ws.set(None);
         page_resource.refetch();
     });
 
-    let handle_context_edit =
-        Callback::new(move |data: (Context, Map<String, Value>)| {
-            let (context, overrides) = data;
-            let conditions = extract_conditions(&Value::Object(context.condition.into()));
+    let on_context_edit = Callback::new(move |data: (Context, Map<String, Value>)| {
+        let (context, overrides) = data;
+        let conditions =
+            Conditions::from_context_json(&context.condition.into()).unwrap();
 
-            set_selected_data.set(Some(Data {
-                context: conditions,
-                overrides: overrides.into_iter().collect::<Vec<(String, Value)>>(),
-            }));
-            set_form_mode.set(Some(FormMode::Edit));
+        selected_context_ws.set(Some(Data {
+            context: conditions,
+            overrides: overrides.into_iter().collect::<Vec<(String, Value)>>(),
+        }));
+        set_form_mode.set(Some(FormMode::Edit));
 
-            open_drawer("context_and_override_drawer");
-        });
+        open_drawer("context_and_override_drawer");
+    });
 
-    let handle_context_clone =
-        Callback::new(move |data: (Context, Map<String, Value>)| {
-            let (context, overrides) = data;
-            let conditions = extract_conditions(&Value::Object(context.condition.into()));
+    let on_context_clone = Callback::new(move |data: (Context, Map<String, Value>)| {
+        let (context, overrides) = data;
+        let conditions =
+            Conditions::from_context_json(&context.condition.into()).unwrap();
 
-            set_selected_data.set(Some(Data {
-                context: conditions,
-                overrides: overrides.into_iter().collect::<Vec<(String, Value)>>(),
-            }));
-            set_form_mode.set(Some(FormMode::Create));
+        selected_context_ws.set(Some(Data {
+            context: conditions,
+            overrides: overrides.into_iter().collect::<Vec<(String, Value)>>(),
+        }));
+        set_form_mode.set(Some(FormMode::Create));
 
-            open_drawer("context_and_override_drawer");
-        });
+        open_drawer("context_and_override_drawer");
```

Comment:
Can we create ENUMs for these strings , it's a little dangerous to keep it as strings

Reply:
Drawers are going away in my next PR, so this will be redundant work.